core/health/rest/src/main/java/org/opennms/core/health/rest/HealthCheckRestService.java

```diff
@@ -42,10 +43,5 @@
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    Response getHealth(@QueryParam("t") @DefaultValue("5000") int timeoutInMs);
-
-    @GET
-    @Path("/tags/{tag}")
-    @Produces(MediaType.APPLICATION_JSON)
-    Response getHealthByTag(@QueryParam("t") @DefaultValue("5000") int timeoutInMs, @PathParam("tag") String tag);
+    Response getHealth(@QueryParam("t") @DefaultValue("5000") int timeoutInMs,@QueryParam("tag") @DefaultValue("") List<String> tags);
```

Comment:
Using UriInfo would be more appropriate.

Reply:
why?

Reply:
Hopefully I'm not missing anything here. I see both way would work. But if we use the UriInfo, then that means user can input anything during the runtime so that user would have more chance to fail because the code doesn't handle it. If we define the query parameter individually, user would know what we specifically want them to input, which would be more intuitive and helpful for the user. Below is the screenshot I've had so far. There could be more enhancements that can be done in this feature, I'm not sure if we want them all in this jira/pull request or we want to break them down and finished by steps. 
![Screen Shot 2021-06-08 at 4 11 12 PM](https://user-images.githubusercontent.com/4617120/121250966-59020c00-c874-11eb-867b-45ece8fecc0c.png)

Reply:
Yes, I've noticed this in the CI test also. It is because minion JAX RS doesn't support this. I changed the List<String> to String[], then it turned out minion jaxrx supports this, but OpenNMS JAX RS doesn't support it. Can you check the difference when you enabled the REST in minion? It seems the minion is not consistent with OpenNMS, should because of a different version or different setting. Minion and OpenNMS should have the same expected behavior with the same code since they share the same Karaf feature. 
I can also try to use UriInfo to see if they behave the same.

Reply:
UriInfo works for OpenNMS but failed for Minion on my machine this morning, but it seems to work just now when I try it again. I pushed the code and let's give it a try in CI.

Reply:
Great news, the Ci is passed :).  If we are good with the current implementation, I'm going to start checking the test code and docs.